### PR TITLE
Read object through io::RandomAccessFile

### PR DESCRIPTION
**Pull request by sduzh**
## Problem Summary(Required) ：

Read object through io::RandomAccessFile

## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] others

### CODE REVIEW

**Comment by sduzh** [comment]
run starrocks_clang-format

File: be/src/env/env_s3.cpp [hunk 1]
```diff
@@ -35,41 +30,33 @@ class S3RandomAccessFile : public RandomAccessFile {
...
     std::string _object_path;
     mutable uint64_t _object_size;
 };
 
-Status S3RandomAccessFile::read(uint64_t offset, Slice* res) const {
-    size_t read_size = 0;
-    RETURN_IF_ERROR(_client->get_object_range(_bucket, _object, offset, res->size, res->data, &read_size));
-    res->size = read_size;
+Status RandomAccessFileWrapper::read(uint64_t offset, Slice* res) const {
+    if (UNLIKELY(offset > std::numeric_limits<int64_t>::max())) return Status::NotSupported("offset overflow");
```
Comment (starrocks-xueli) [inline be/src/env/env_s3.cpp:39]: this seems will never be true.

Comment (sduzh) [reply be/src/env/env_s3.cpp:39]: Do you mean this won't happen in practice?

Comment (starrocks-xueli) [reply be/src/env/env_s3.cpp:39]: if `offset` == `std::numeric_limits<int64_t>::max()`, i think `offset + 1` will be 0?

Comment (starrocks-xueli) [reply be/src/env/env_s3.cpp:39]: i see, it's int64_t not uint64_t.

**Review by starrocks-xupeng — APPROVED** [review]

File: be/src/env/env_s3.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/env/env_s3.cpp:39]: Yes, you're right!

**Comment by sduzh** [comment]
@wyb @dirtysalt PTAL

**Review by dirtysalt — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**